### PR TITLE
Fix tracing

### PR DESCRIPTION
**Pull request by shyamjesal**
- add ZipkinEndpoint environment variable
- add custom span `InvokeWithXDT` in `sdk/golang/source.go`
- add zipkin container in docker-compose.
  - modify fx to poll zipkin endpoint before starting

### CODE REVIEW

File: tests/integration_test.go [hunk 1]
```diff
@@ -125,6 +127,10 @@ func knativeQP(config utils.Config) {
 
 		}
 	}(composedHandler)
+
+	if config.TracingEnabled {
+		composedHandler = otelhttp.NewHandler(composedHandler, "HTTPProxy")
```
Comment (ustiugov) [inline tests/integration_test.go:132]: please let Michal know (i.e., raise and assign an issue) that we need to add this to the tracing library. leave the code here as is

File: user-functions/dQP/go.mod [hunk 2]
```diff
@@ -16,8 +17,9 @@ require (
 	github.com/ease-lab/vhive-xdt/utils v0.0.0-00010101000000-000000000000
 	github.com/ease-lab/vhive/utils/tracing/go v0.0.0-20210623161727-460bac97d8c0
 	github.com/sirupsen/logrus v1.8.1
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.0.0-00010101000000-000000000000
 	golang.org/x/net v0.0.0-20210614182718-04defd469f4e
 	google.golang.org/grpc v1.38.0
 	knative.dev/pkg v0.0.0-20210625194144-4cdacd04734a
-	knative.dev/serving v0.23.1
+	knative.dev/serving v0.24.0
```
Comment (ustiugov) [inline user-functions/dQP/go.mod:24]: careful with that... the vhive repo is still 0.23. maybe update Serving there as well

File: user-functions/fx/fx.go [hunk 3]
```diff
@@ -111,10 +113,33 @@ func main() {
 
 	config := utils.ReadConfig()
 	if *dockerCompose {
+		if config.TracingEnabled {
+			timeout := 1 * time.Second
+			for {
```
Comment (ustiugov) [inline user-functions/fx/fx.go:118]: should we offload that to the tracing library? discuss with Michal

**Review by ustiugov — APPROVED** [review]
the pr can be merged but please raise the following issues before merging this PR

- [x] raise an issue for the NewHandler for Michal in the vhive repo. Also, do we need to support dialing?
- [x] raise an issue for updating Serving fork to 0.24

File: user-functions/fx/fx.go (hunk 3 above)
Comment (shyamjesal) [reply user-functions/fx/fx.go:118]: I don't think so. Docker compose only supports the control of the boot sequence. It does not support waiting for a container to boot up/endpoint to be ready. Such situation will not arrive in Knative integration as the Zipkin pods will be ready beforehand.

File: user-functions/dQP/go.mod (hunk 2 above)
Comment (shyamjesal) [reply user-functions/dQP/go.mod:24]: User functions for dQP and sQP are just to mimic the functionality of Knative and testing. This code is not used by the `que-proxy/dQP`.

**Pull request merged**